Source/CombatExtended/CombatExtended/AmmoInjector.cs

```diff
@@ -61,8 +61,8 @@ public static bool InjectAmmos()
 	            foreach (ThingDef def in DefDatabase<ThingDef>.AllDefsListForReading)
 	            {
 	            	if (def.IsWeapon
-	            	    && (def.generateAllowChance <= 0
-	            	        || def.tradeability == Tradeability.Buyable
+	            	    && (def.generateAllowChance > 0
+	            	        || def.tradeability.TraderCanSell()
```

Comment:
This seems like it would break determination of availability, what's the reasoning behind this change?

Reply:
That method means tradeability == Buyable || tradeability == All

Reply:
And why generateAllowChance should less than or equal 0 to be a weapon?

Reply:
Good change.

These changes by kpqi5858 are as they should be. The code present right now is actually wrong, because it only picks weapons which aren't generatable.